apicast/src/oauth/keycloak.lua

```diff
@@ -248,7 +262,8 @@ function _M:get_token(service, client)
   -- call Keycloak authorize
   local url = self.config.token_url
 
-  local res = http_client.post(url, req_body)
+  local headers = ngx.req.get_headers()
+  local res = http_client.post(url, ngx.req.get_post_args(), { headers = headers })
```

Comment:
There is a test that verifies that headers are passed at all: https://github.com/3scale/apicast/pull/336/files/8cd98a1f55125a836cd94603cca2223460d0d716#diff-bc82c1966e003ef46c8dc5a71bb64c5d
Because they were not before, so the test would fail.
Now, regarding all headers VS whitelist, for now I really only need the `Authorization` header, but I am not sure if there can be some other headers that Keycloak may be expecting.
We can limit it to `Authorization` for now, and test that others are not proxy passed (although not sure how to do it with test backend).